common/thorhelper/roxiehelper.cpp

```diff
@@ -1648,6 +1646,31 @@ int readHttpHeaderLine(IBufferedSocket *linereader, char *headerline, unsigned m
     return bytesread;
 }
 
+void parseHttpParameterString(IProperties *p, const char *str)
+{
+    while (*str)
+    {
+        StringBuffer s, prop, val;
+        while (*str && *str != '&' && *str != '=')
+            s.append(*str++);
+        appendDecodedURL(prop, s.trim());
+        if (!*str || *str == '&')
+            val.set("1");
+        else
+        {
+            s.clear();
+            str++;
+            while (*str && *str != '&')
+                s.append(*str++);
+            appendDecodedURL(val, s.trim());
+        }
+        if (prop.length())
+            p->setProp(prop, val);
+        if (*str)
+            str++;
+    }
+}
```

Comment:
Right.  As it is this is old code that I would have to look at beefing up
and abstracting before claiming it could be used general purpose for
parsing URLs beyond the needs of mapping parameters to queries.

On Tue, Mar 22, 2016 at 3:17 PM, Gordon Smith <notifications@github.com>
wrote:

> In common/thorhelper/roxiehelper.cpp
> <https://github.com/hpcc-systems/HPCC-Platform/pull/8369#discussion_r57051110>
> :
>
> > +        if (!*str || *str == '&')
> > +            val.set("1");
> > +        else
> > +        {
> > +            s.clear();
> > +            str++;
> > +            while (*str && *str != '&')
> > +                s.append(*str++);
> > +            appendDecodedURL(val, s.trim());
> > +        }
> > +        if (prop.length())
> > +            p->setProp(prop, val);
> > +        if (*str)
> > +            str++;
> > +    }
> > +}
>
> Correct - but I can't think of any scenario where you would need to
> support it. But if this is to become a generic URL parser then it would be
> worth adding.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcc-systems/HPCC-Platform/pull/8369/files/15c2730e4177460c9bb7792c5a572647965244f3#r57051110>
>